mmv1/products/compute/api.yaml

```diff
@@ -7357,13 +7356,34 @@ objects:
           selected availability domain will be provided to the Partner via the
           pairing key so that the provisioned circuit will lie in the specified
           domain. If not specified, the value will default to AVAILABILITY_DOMAIN_ANY.
+        values:
+          - :AVAILABILITY_DOMAIN_1
+          - :AVAILABILITY_DOMAIN_2
+          - :AVAILABILITY_DOMAIN_ANY
+      - !ruby/object:Api::Type::NestedObject
+        name: 'partnerMetadata'
+        description: |
+          Informational metadata about Partner attachments from Partners to display
+          to customers. Output only for for PARTNER type, mutable for PARTNER_PROVIDER,
+          not available for DEDICATED.
+        properties:
+          - !ruby/object:Api::Type::String
+            name: partnerName
+            description: |
+              Plain text name of the Partner providing this attachment. This value may be validated to match approved Partner values.
+          - !ruby/object:Api::Type::String
+            name: interconnectName
+            description: |
+              Plain text name of the Interconnect this attachment is connected to, as displayed in the Partner's portal. For instance "Chicago 1". This value may be validated to match approved Partner values.
+          - !ruby/object:Api::Type::String
+            name: portalUrl
+            description: |
+              URL of the Partner's portal for this Attachment. Partners may customise this to be a deep link to the specific resource on the Partner portal. This value may be validated to match approved Partner values.
       - !ruby/object:Api::Type::String
         name: 'pairingKey'
+        input: true
         description: |
-          [Output only for type PARTNER. Not present for DEDICATED]. The opaque
-          identifier of an PARTNER attachment used to initiate provisioning with
-          a selected partner. Of the form "XXXXX/region/domain"
-        output: true
+          [Output only for type PARTNER. Input only for PARTNER_PROVIDER. Not present for DEDICATED]. The opaque identifier of an PARTNER attachment used to initiate provisioning with a selected partner. Of the form "XXXXX/region/domain"
```

Comment:
Let me get this straight:

- if the interconnect is of type PARTNER, this value is output by the API
- if the interconnect is of type PARTNER_PROVIDER, this value is an input to the API but is not returned
- if the interconnect is of type DEDICATED, this is neither an input nor an output

Reply:
Correct. Creating an interconnect as a PARTNER generates the pairing key to be consumed. An PARTNER_PROVIDER interconnect will consume the pairing key to complete the pairing and enable the interconnect. 

Dedicated interconnects do not need to go through the pairing steps.